bundle/deploy/terraform/convert.go

```diff
@@ -79,222 +78,64 @@ func BundleToTerraformWithDynValue(ctx context.Context, root dyn.Value) (*schema
-				if config.Resources.Schemas == nil {
-					config.Resources.Schemas = make(map[string]*resources.Schema)
-				}
-				cur := config.Resources.Schemas[resource.Name]
-				if cur == nil {
-					cur = &resources.Schema{ModifiedStatus: resources.ModifiedStatusDeleted}
-				}
-				cur.ID = instance.Attributes.ID
-				config.Resources.Schemas[resource.Name] = cur
-			case "databricks_volume":
-				if config.Resources.Volumes == nil {
-					config.Resources.Volumes = make(map[string]*resources.Volume)
-				}
-				cur := config.Resources.Volumes[resource.Name]
-				if cur == nil {
-					cur = &resources.Volume{ModifiedStatus: resources.ModifiedStatusDeleted}
-				}
-				cur.ID = instance.Attributes.ID
-				config.Resources.Volumes[resource.Name] = cur
-			case "databricks_cluster":
-				if config.Resources.Clusters == nil {
-					config.Resources.Clusters = make(map[string]*resources.Cluster)
-				}
-				cur := config.Resources.Clusters[resource.Name]
-				if cur == nil {
-					cur = &resources.Cluster{ModifiedStatus: resources.ModifiedStatusDeleted}
-				}
-				cur.ID = instance.Attributes.ID
-				config.Resources.Clusters[resource.Name] = cur
-			case "databricks_dashboard":
-				if config.Resources.Dashboards == nil {
-					config.Resources.Dashboards = make(map[string]*resources.Dashboard)
-				}
-				cur := config.Resources.Dashboards[resource.Name]
-				if cur == nil {
-					cur = &resources.Dashboard{ModifiedStatus: resources.ModifiedStatusDeleted}
-				}
-				cur.ID = instance.Attributes.ID
-				config.Resources.Dashboards[resource.Name] = cur
-			case "databricks_app":
-				if config.Resources.Apps == nil {
-					config.Resources.Apps = make(map[string]*resources.App)
-				}
-				cur := config.Resources.Apps[resource.Name]
-				if cur == nil {
-					cur = &resources.App{ModifiedStatus: resources.ModifiedStatusDeleted}
-				} else {
-					// If the app exists in terraform and bundle, we always set modified status to updated
-					// because we don't really know if the app source code was updated or not.
-					cur.ModifiedStatus = resources.ModifiedStatusUpdated
-				}
-				cur.ID = instance.Attributes.Name
-				config.Resources.Apps[resource.Name] = cur
-			case "databricks_secret_scope":
-				if config.Resources.SecretScopes == nil {
-					config.Resources.SecretScopes = make(map[string]*resources.SecretScope)
-				}
-				cur := config.Resources.SecretScopes[resource.Name]
-				if cur == nil {
-					cur = &resources.SecretScope{ModifiedStatus: resources.ModifiedStatusDeleted}
-				}
-				cur.ID = instance.Attributes.Name
-				config.Resources.SecretScopes[resource.Name] = cur
-			case "databricks_permissions":
-			case "databricks_grants":
-			case "databricks_secret_acl":
-				// Ignore; no need to pull these back into the configuration.
-			default:
-				return fmt.Errorf("missing mapping for %s", resource.Type)
-			}
+func ensureMap(v dyn.Value, path dyn.Path) (dyn.Value, error) {
+	item, _ := dyn.GetByPath(v, path)
+	if !item.IsValid() {
```

Comment:
Panic or error if it is not a map to be defensive?

Reply:
If any parents are of invalid type (should not be possible), SetByPath later will return an error. This seems enough.